automated install/basic-install.sh

```diff
@@ -310,7 +309,21 @@ elif is_command rpm ; then
             # exit the installer
             exit
         fi
-        # on CentOS we need to add the EPEL repository to gain access to Fedora packages
+        # php-json is not required on CentOS 7 as it is already compiled into php
+        # verifiy via `php -m | grep json`
+        if [[ $CURRENT_CENTOS_VERSION -eq 7 ]]; then
+            # create a temporary array as arrays are not designed for use as mutable data structures
+            CENTOS7_PIHOLE_WEB_DEPS=()
+            for i in "${!PIHOLE_WEB_DEPS[@]}"; do
+                if [[ ${PIHOLE_WEB_DEPS[i]} != "php-json" ]]; then
+                    CENTOS7_PIHOLE_WEB_DEPS+=( "${PIHOLE_WEB_DEPS[i]}" )
+                fi
+            done
+            # re-assign the clean dependency array back to PIHOLE_WEB_DEPS
+            PIHOLE_WEB_DEPS=("${CENTOS7_PIHOLE_WEB_DEPS[@]}")
+            unset CLEAN_PIHOLE_WEB_DEPS
```

Comment:
Shouldn't this be the following?
```suggestion
            unset CENTOS7_PIHOLE_WEB_DEPS
```

Reply:
Ah yes, I had re-named the variable before the pushing and missed that one!   
I am surprised that CI did not catch an unset of an un-used variable.. 
I have updated, squashed, and cleaned up my commit message.

Thanks!